runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingModeExecutionContext.java

```diff
@@ -152,7 +151,9 @@ public StreamingModeExecutionContext(
       MetricsContainerRegistry<StreamingStepMetricsContainer> metricsContainerRegistry,
       DataflowExecutionStateTracker executionStateTracker,
       StreamingModeExecutionStateRegistry executionStateRegistry,
-      long sinkByteLimit) {
+      StreamingEnginePipelineConfigManager configManager,
```

Comment:
Do we need/want this to possibly change limits between calls?
The single OperationalLimits would avoid synchronization on hot-path and possible bugs if callers of the getmax methods don't expect the value to change (between start/finish)

If we do for some reason, how about having this take a Supplier<StreamingEnginePipelineConfig>?
We don't want this to call any of the other methods on the manager, and it also will simplify testing or perhaps appliance since we can just pass in a static supplier.

Reply:
Good point about keeping the config consistent per bundle, fixed it.

> how about having this take a Supplier?

Extracted the public methods to a interface `StreamingEnginePipelineConfigManager` and added a different implementation for Appliance/Testing. I like specific interfaces compared to common interfaces like Consumer/Supplier. The common interfaces makes it harder to find references in IDE.